### ISSUE TITLE
harmonize flags

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#669]
Please refer to: https://github.com/neuropoly/spinalcordtoolbox/wiki/flags
Important: if you make change to a flag, please use deprecated mode and indicate in this thread.
If not possible to use deprecated, please list potential conflicts for each file.
If no parser in the script, add it.

**Sara**
- [x] sct_apply_transfo 
 - "-c" changed to "-crop"
- [x] sct_average_data_within_mask
 - added parser
 - "-t" was used to number of volume to use if 4D. Changed it to "-nvol".
- [x] sct_check_atlas_integrity
 - added parser
 - "-m" was used for GM segmentation. Changed it to "-gm"
 - "-t" was use for atlas threshold. Changed it to "-thr"
 - "-g" was use for GM threshold. Changed it to "-thrgm" (to match with "-thr")
- [x] sct_check_dependences
 - added parser
 - "-l" was used for log file option. Changed it to "-log"
- [x] sct_compute_ernst_angle
 - added get_parser() method.
 - "-d" was used for display option. Changed it to "-v"
- [x] sct_compute_hausdorff_distance
 - added get_parser() method.
 - "-t" was used for thinning. Changed it to "-thinning"
 - "-r" was used for second input (reference). Changed it to "-d"
- [x] sct_compute_mtr
 - added parser
 - changed "-i" and "-j" to "-mt0" and "mt1" to be more explicit.
- [x] sct_concat_transfo
 - added parser
- [x] sct_convert (already up to date with flags convention)
- [x] sct_segment_graymatter
 - "-l" was for vertebral level file. Changed to "-vert"
- [x] sct_register_to_template
 - '-p' replaced by '-param'
- [x] sct_resample
 - added get_parser() function
- [x] sct_smooth_spinalcord
 - added parser
 - changed flag '-s' to '-smooth' **WARNING: -s is not deprecated to -smooth**
 - changed "-c" (centerline or seg) to "-s"
- [x] sct_straighten_spinalcord
 - changed flag '-p' to '-pad'
 - changed flag '-params' to '-param'
 - changed flag "-c" to "-s"
- [x] sct_testing
 - added parser
- [x] sct_warp_template
 - changed flag -o to -ofolder

**Simon**
- [x] sct_create_mask
 - added parser
 - "-m" was for method. Changed to "-p"
 - "-s" was for size, changed for "-size"
- [x] sct_crop_image
 - added get_parser function.
- [x] sct_dice_coefficient
  - did a warper for the sct_dice_coefficient binary (now isct_dice...)
  - input files now have flags -i and -d **WARNING: you will need to change this in your scripts, there was no flags before so there is no deprecated option ...**
- [x] sct_dmri_compute_dti (already up to date with flags convention)
- [x] sct_dmri_concat_bvals (already up to date with flags convention)
- [x] sct_dmri_concat_bvecs (already up to date with flags convention)
- [x] sct_dmri_get_bvalue
 - added parser
- [x] sct_dmri_moco
 - added parser
 - '-b' was for bvecs file, changed to '-bvec'
 - '-p' was for param, changed to '-param'
 - '-t' was for an otsu threshold, changed to '-thr'
 - '-o' was for output folder, changed to '-ofolder'

**Benjamin**
- [x] sct_dmri_separate_b0_and_dwi
 - "-m" was used for bvals. Changed it to "-bval".
 - "-b" was used for bvecs. Changed it to "-bvec".
 - "-o" was used for output folder. Changed it to "-ofolder".
- [x] sct_dmri_transpose_bvecs
 - "-i" was used for bvec file. Changed it to "-bvec". "-i" is now deprecated.
- [x] sct_download_data
- [x] sct_extract_metric
 - No parser.
 - '-p' changed to '-param' 
- [x] sct_flatten_sagittal
 - No parser.
 - "-c" changed to "-s"
 - "-s" changed to "-x" **WARNING: -s is not depredated to -x**
- [x] sct_fmri_compute_tsnr
 - added get_parser function.
- [x] sct_fmri_moco
 - No parser.
 - '-p' changed to '-param'
- [x] sct_get_centerline
 - "-t" option was used for contrast. It is now deprecated by "-c".
 - "-p" option was used for a file containing a point. It is now deprecated by "-point".

**Olivier**
- [x] sct_image
- [x] sct_label_utils
 - "-r" was used to reference an image. Now deprecated by "-ref"
 - "-level" was used for vertebral level. Now deprecated by "-vert"
 - "-t" --> changed for -p (julien)
- [x] sct_label_vertebrae
 - "-t" was used for processes, now using -p
 - "-seg" for segmentation file is now "-s"
- [x] sct_maths
- [x] sct_process_segmentation
 - "-l" switched to "-vert" for vertebral levels
 - "-s" replaced by "-size"
 - No Parser
- [x] sct_propseg
 - "-t" option is for contrast
- [x] sct_register_multimodal
 - "-p" was used for params, now -param

**Comment by SaraDupont** [comment]
Do it on branch **harmonize_flags**

**Comment by benjamindeleener** [comment]
If you need to create a new parser, don't forget to use the get_parser function standard. Take as exemple  sct_straighten_spinalcord

**Comment by jcohenadad** [comment]
@benjamindeleener could you give an example code of how to use the "deprecated" option?  we all need to use it (whenever possible)

**Comment by benjamindeleener** [comment]
See ```sct_dmri_separate_b0_and_dwi.py```:
```python
parser.add_option(name='-o',
                  type_value='folder_creation',
                  description='Output folder.',
                  mandatory=False,
                  default_value='./',
                  deprecated_by='-ofolder')
```

The important thing is to use deprecated_by when the option is replaced so that the parser will use the new option if the user provides the old one.

**Comment by jcohenadad** [comment]
so just to be sure: in your example the old is "-o" and the new is "-ofolder", is that correct?

**Comment by benjamindeleener** [comment]
@jcohenadad Yes

**Comment by SaraDupont** [comment]
Do we want to modify sct_dice_coefficient ? (since it's a binary file ...) Or we can make a warper for the binary with a parser and everything ?

**Comment by jcohenadad** [comment]
yes we should make a wrapper. Can you do that?
thanks!

**Comment by SaraDupont** [comment]
OK ! I'll do that after we merge master into harmonize_flags since I will need to refactor every use of sct_dice_coefficient in the code (for now there are no flags for the inputs in dice_coeff (see doc below) so I can't use the deprecated option.) 

![screen shot 2015-11-24 at 6 10 28 pm](https://cloud.githubusercontent.com/assets/5857909/11383559/e3bb4eee-92d6-11e5-9009-c49806a237d2.png)

**Comment by jcohenadad** [comment]
ok, i’ll work on the merge tmr morning (i don’t have time to do it today)

--
Julien Cohen-Adad, PhD
Assistant Professor, Polytechnique Montreal
Associate Director, Functional Neuroimaging Unit, University of Montreal
Canada Research Chair in Quantitative Magnetic Resonance Imaging
Phone: 514 340 5121 (office: 2264);  Skype: jcohenadad
Web: www.neuro.polymtl.ca

**Comment by SaraDupont** [comment]
OK, no problem !
Just to be sure, for sct_warp_template, we decided to keep '-s' flag to warp spinal level, right ?

**Comment by jcohenadad** [comment]
yes we did

**Comment by jcohenadad** [comment]
i just noticed something: we cannot use "mandatory=True" in a deprecated parameter. So it should be changed for "mandatory=False".
@benjamindeleener do you confirm?

**Comment by benjamindeleener** [comment]
For deprecated options, the mandatory parameters is not used so you can put whatever you want (normally). If it fails because it requires the option, just put it to mandatory=False so we don't have any problems.

**Comment by jcohenadad** [comment]
fixed in ed86e0fc119e86d28f61fdb4164b6010cd62b0a7
